### ISSUE TITLE
Intl.DateTimeFormat has incorrect ICU format for timeZoneName: 'short'

### ISSUE THREAD

**Issue by davelyon** [facebook/hermes#661]
## Bug Description

Hermes 0.10.0 on Android devices that can use ICU formatting will always produce time zone offset values, rather than potentially more localized / human readable time zone names. For example, given a time zone value of `America/Phoenix` the formatted value of the time zone name will always be `GMT-7` rather than the more user-recognizable `MST`. 

This is because the format used with the underlying ICU library is `O` (Time Zone: short localized GMT
) for short and `VV` (Time Zone: long time zone ID) for long. See the chart here for reference: https://unicode-org.github.io/icu/userguide/format_parse/datetime/#date-field-symbol-table

Here's the relevant code from this repository: https://github.com/facebook/hermes/blob/c2cd9e385a922f486a55e6bd70db2032f78379ed/lib/Platform/Intl/java/com/facebook/hermes/intl/IPlatformDateTimeFormatter.java#L377-L386

- [x] I have run `gradle clean` and confirmed this bug does not occur with JSC

Hermes version: 0.10.0
React Native version (if any): 0.66, 0.67
OS version (if any): Android 12
Platform (most likely one of arm64-v8a, armeabi-v7a, x86, x86_64):

## Code example:

```js
const formatter = new Intl.DateTimeFormat([ 'en-US'], {   hour: 'numeric',   minute: '2-digit', timeZone: 'America/Phoenix', timeZoneName: 'short' })

formatter.formatToParts(new Date())

/*
Expected:
[
 // truncated
  { type: 'timeZoneName', value: 'MST' }
]

Actual:
[
 // truncated
  { type: 'timeZoneName', value: 'GMT-7' }
]

*/

```

## The Expected Behavior

The latest ECMA 402 spec expands the values of `timeZoneName` to allow for more specific behavior, where the current behavior of the Hermes ICU formatter most closely matches `shortOffset` rather than `short`. I would propose that the correct behavior of Hermes, given it does not currently support the more recent spec, should be to use `zzz` (Time Zone: specific non-location) which would allow it to produce values like `EST` or `PDT` where possible.

I would be happy to create a pull request with this change and relevant test cases if this is acceptable.

**Comment by neildhar** [comment]
Apologies that it has taken so long for us to respond. We would welcome a PR if you're willing to submit one.

**Comment by arys** [comment]
@davelyon is there any workaround for this?

**Comment by statico** [comment]
This would be wonderful to fix since this causes inconsistency between iOS and Android.

**Comment by neildhar** [comment]
I believe this is a duplicate of https://github.com/facebook/hermes/issues/1001, which was fixed in 6d478c2e3d07e820d2178e9708ab964f50f943ac
